cmd/invoke.go

```diff
@@ -0,0 +1,403 @@
+
+	Invocation Target
+	  The Function instance to invoke can be specified using the --target flag
+	  which accepts the values "local", "remote", or <URL>.  By default the
+	  local Function instance is chosen if running (see {{.Prefix}}func run).
+	  To explicitly target the remote (deployed) Function:
+	    {{.Prefix}}func invoke --target=remote
+	  To target an arbitrary endpont, provide a URL:
+	    {{.Prefix}}func invoke --target=https://myfunction.example.com
+
+	File Submission
+	  Providing a filename in the --file flag will base64 encode its contents
+	  as the "data" parameter sent to the Function.  The value of --content-type
+	  should be set to the type from the source file.  For example, the following
+	  would send a JPEG base64 encoded in the "data" POST parameter:
+	    {{.Prefix}}func invoke --file=example.jpeg --content-type=image/jpeg
+
+	Message Format
+	  By default Functions are sent messages which match the invocation format
+	  of the template they were created using; for example "http" or "cloudevent".
+	  To override this behavior, use the --format (-f) flag.
+	    {{.Prefix}}func invoke -f=cloudevent -t=http://my-sink.my-cluster
+
+EXAMPLES
+
+	o Invoke the default (local or remote) running Function with default values
+	  $ {{.Prefix}}func invoke
+
+	o Run the Function locally and then invoke it with a test request:
+	  (run in two terminals or by running the first in the background)
+	  $ {{.Prefix}}func run
+	  $ {{.Prefix}}func invoke
+
+	o Deploy and then invoke the remote Function:
+	  $ {{.Prefix}}func deploy
+	  $ {{.Prefix}}func invoke
+
+	o Invoke a remote (deployed) Function when it is already running locally:
+	  (overrides the default behavior of preferring locally running instances)
+	  $ {{.Prefix}}func invoke --target=remote
+
+	o Specify the data to send to the Function as a flag
+	  $ {{.Prefix}}func invoke --data="Hello World!"
+
+	o Send a JPEG to the Function
+	  $ {{.Prefix}}func invoke --file=example.jpeg --content-type=image/jpeg
+
+	o Invoke an arbitrary endpoint (HTTP POST)
+		$ {{.Prefix}}func invoke --target="https://my-event-broker.example.com"
+
+	o Invoke an arbitrary endpoint (CloudEvent)
+		$ {{.Prefix}}func invoke -f=cloudevent -t="https://my-event-broker.example.com"
+
+`,
+		SuggestFor: []string{"emit", "emti", "send", "emit", "exec", "nivoke", "onvoke", "unvoke", "knvoke", "imvoke", "ihvoke", "ibvoke"},
+		PreRunE:    bindEnv("path", "format", "target", "id", "source", "type", "data", "content-type", "file", "confirm", "namespace"),
+	}
+
+	// Flags
+	cmd.Flags().StringP("path", "p", cwd(), "Path to the Function which should have its instance invoked (Env: $FUNC_PATH)")
+	cmd.Flags().StringP("format", "f", "", "Format of message to send, 'http' or 'cloudevent'.  Defaults is to auto-choose, but is required for direct invocation of remote CloudEvent Functions by URL, or for Functions created prior. (Env: $FUNC_FORMAT)")
```

Comment:
Prior to.... _oh look a kitty_

I got distracted looking up the version number and never finished my sentence.  I think that perhaps, since we are pre 1.0 to begin with, (and I added clear instructions for specifying message type in examples), I will just omit this caviat as it will perhaps cause more confusion than it will prevent!